lib/honeycomb/client.rb

```diff
@@ -71,6 +63,19 @@ def start_span(name:, serialized_trace: nil, **fields)
       rescue StandardError => e
         current_span.add_field("error", e.class.name)
         current_span.add_field("error_detail", e.message)
+
+        if error_backtrace_limit > 0
+          # Honeycomb by default escapes things like brackets and shows the
+          # unicode sequence. So <main> becomes \u003cmain\u003e. Given the
+          # confusion that could cause in readability, we replace < and > with
+          # a single quote.
+          backtrace = e.backtrace[0...error_backtrace_limit]
```

Comment:
Is it possible to know how many levels the back trace has? If we know the limit is >= number of levels, we will have a complete backtrace not a partial.

Reply:
It is possible. `e.backtrace` is an array, so `e.backtrace.length` or `.size` would tell us how many levels there are. Are you picturing changing the name of the field being added–`error_backtrace_partial` or `error_backtrace`–based on whether the backtrace length is less than or equal to the limit?

Reply:
I'd be more inclined to leave this as the same name no matter what (and maybe just `error_backtrace`) but introduce two additional columns: `error_backtrace_limit` and `error_backtrace_length`. With those two, someone could make a derived column of, like, `error_backtrace_is_complete` or something by doing a quick comparison.

But maybe adding 3 new columns with 1 PR is a bit much.

Reply:
@robbkidd and I have just discussed and came to the same conclusion that event fields for the backtrace, length and limit would be better as it would provide consistent field names.

We're like your suggestions of:
```
error_backtrace
error_backtrace_length
error_backtrace_limit
```

If you're happy to update to add them that would great, we can add them.

Reply:
I did `error_backtrace_total_length` but am happy to undo that little bit. 

https://github.com/honeycombio/beeline-ruby/pull/160/commits/7d3eb6f187783664db2b8b16f591342162bed203